File: MiniMarket/MiniMarket/Model/VoucherDto.vb
Public Class VoucherDto
    Private _voucherId As Integer
    Public Property VoucherId() As Integer
        Get
            Return _voucherId
        End Get
        Set(ByVal value As Integer)
            _voucherId = value
        End Set
    End Property

    Private _voucherCode As String
    Public Property VoucherCode() As String
        Get
            Return _voucherCode
        End Get
        Set(ByVal value As String)
            _voucherCode = value
        End Set
    End Property

    Private _nominal As Decimal
    Public Property Nominal() As Decimal
        Get
            Return _nominal
        End Get
        Set(ByVal value As Decimal)
            _nominal = value
        End Set
    End Property

    Private _status As String
    Public Property Status() As String
        Get
            Return _status
        End Get
        Set(ByVal value As String)
            _status = value
        End Set
    End Property

    Private _validFrom As DateTime
    Public Property ValidFrom() As DateTime
        Get
            Return _validFrom
        End Get
        Set(ByVal value As DateTime)
            _validFrom = value
        End Set
    End Property

    Private _validTo As DateTime?
    Public Property ValidTo() As DateTime?
        Get
            Return _validTo
        End Get
        Set(ByVal value As DateTime?)
            _validTo = value
        End Set
    End Property

    Private _memberId As Integer
    Public Property MemberId() As Integer
        Get
            Return _memberId
        End Get
        Set(ByVal value As Integer)
            _memberId = value
        End Set
    End Property

    Private _noFaktur As String
    Public Property No_Faktur() As String
        Get
            Return _noFaktur
        End Get
        Set(ByVal value As String)
            _noFaktur = value
        End Set
    End Property
End Class

File: src/Application/Vouchers/Queries/VoucherDto.cs
using AutoMapper;
using UMApplication.Application.Common.Mappings;
using UMApplication.Domain.Entities;
using System;
using UMApplication.Domain.Extension;

namespace UMApplication.Application.Vouchers.Queries
{
    public class VoucherDto : IMapFrom<Voucher>
    {
        public int VoucherId { get; set; }
        public string VoucherCode { get; set; }
        public string Status { get; set; }
        public string No_Faktur { get; set; }
        public string Nominal { get; set; }
        public DateTime ValidFrom { get; set; }

        public string ValidFromString { get; set; }
        public DateTime? ValidTo { get; set; }

        public string ValidToString { get; set; }
        public DateTime? UsageDate { get; set; }

        public string UsageDateString { get; set; }

        public void Mapping(Profile profile)
        {
            profile.CreateMap<Voucher, VoucherDto>()
                  .ForMember(m => m.Nominal, opt => opt.MapFrom(s => s.Nominal.ToString("N0")))
                  .ForMember(m => m.ValidFromString, opt => opt.MapFrom(s => s.ValidFrom.ToShortDateString()))
                  .ForMember(m => m.ValidToString, opt => opt.MapFrom(s => s.ValidTo.ToDateString("dd-mm-yy")))
                  .ForMember(m => m.UsageDateString, opt => opt.MapFrom(s => s.UsageDate.ToDateString("dd-mm-yy")))
                  ;
        }
    }
}
